common/core/src/main/java/com/vmware/connectors/common/payloads/response/CardAction.java

```diff
@@ -403,4 +403,49 @@ public CardAction build() {
             return completedAction;
         }
     }
+
+    public String hash() {
+        final StringBuilder result = new StringBuilder();
+
+        result.append(primary);
+
+        if (StringUtils.isNotBlank(label)) {
```

Comment:
All of these null/empty checks are going to be a problem.  Take an object that looks like this:

```javascript
{
    x: 'a',
    y: 'b',
    z: 'c'
}
```
`sha1('abc')`

```javascript
{
    x: 'a',
    y: null,
    z: 'bc'
}
```
is also `sha1('abc')`

Reply:
Similar to my concerns about toString, string concat (alone) is going to have some issues.

Quick googling didn't turn up anything I liked for computing a hash on objects.

We may end up having to write something that calculates this more generically ourselves (we could do something with serialization, but that introduces its own headaches).

Something that would be nice would be something that takes in varargs like `Objects.hash` (but not that, b/c the details of that are not appropriate for this -- maybe we can leverage it, but we have to be careful) or something uses annotations like `HashCodeBuilder.reflectionHashCode` (again, hashCode is not what we want; a hard-coded hashCode is a valid hashCode and we clearly wouldn't want that).

Reply:
I looked at the implementations of ```Objects.hash``` and ```HashCodeBuilder.reflectionHashCode```.

```Objects.hash``` just calculates the hash code of each object in the array and sum it up.
```HashCodeBuilder.reflectionHashCode``` is similar to the above one but it uses reflection to access the member fields and calculate hash code.

I tried to google about generating sha-1 for objects in an efficient. I did not find any relevant ideas or example. So we have to come with our solution.

One solution is adding an empty space or some separator before appending with the next field. For example, 

{
    x: 'a',
    y: 'b',
    z: 'c'
}

Appending separator would result: ```sha1(a b c)```

{
    x: 'a',
    y: null,
    z: 'bc'
}
  
Appending separator would result: ```sha1(a bc)```

Now this solution would avoid creating same hash for different objects.

We cannot remove ```null`` check, since ```DigestUtils.sha1Hex(null)``` is going to throw ```NullPointerException```. 

Please let me know the idea of adding a separator after each field and I will change the code accordingly.